### ISSUE TITLE
Allow msbot to sign a .bot file using a secret provided by the developer

### ISSUE THREAD

**Issue by rob-derosa** [microsoft/botbuilder-tools#775]
## Tool
Name: msbot

## Describe the problem
If a developer needs to update the .bot file, say to add/modify a config section, it needs to be decrypted first so it can be edited in plain text. Then it needs to be re-encrypted before being committed. This forces a rotation of the secret which means every developer on the team working on the bot must update the secret in their .env/appsettings.json files each time this happens. 

## Describe the solution you'd like
This could be mitigated if msbot allowed the user to provide a secret when encrypting. Something like:
msbot secret --new --secret XXXXXXXXXXXXXXXXX

basically the same args as the --clear command

## Additional context
The reason I need to modify the .bot file by hand are for a few reasons:
1) The Bot Emulator doesn't include the subscription key for QnA Maker config sections so dispatch doesn't work when using the .bot file as a source. Thus, you have to edit the .bot file by hand to add it in.

2) The Bot Emulator errs out when attempting to sign into Azure for certain AD tenants. I was onsite with Intel in Costa Rica and none of them were able to authenticate with Azure in the Emulator to pull in LUIS/QnA Maker services.

[enhancement]

**Comment by vishwacsena** [comment]
We need a security review on this one before we implement.

**Comment by martinkearn** [comment]
Just for context, I think the idea is that all changes to the .bot file are intended to happen via the MSBOT CLI (or emulator which uses MSBOT I believe), not by editing the file in a text editor. MSBOT will not rotate the secret when you add sections, services etc I don't think. 

So "_If a developer needs to update the .bot file_", they should use the MSBOT CLI.

Not saying that is right, just that if the BOT file is updated the way it is intended, the secret remains intact.

**Comment by lynn-orrell** [comment]
To add another bit of information, the Bot Framework Emulator will allow you to change values in place and re-encrypt with the same key. This should be allowed from the CLI as well.

**Comment by Kaiqb** [comment]
We have deprecated the use of .bot files. The deployment topics were updated to reflect this change.
